plugins/modules/net_tools/nmcli.py

```diff
@@ -1105,7 +1121,17 @@ def remove_connection(self):
         return self.execute_command(cmd)
 
     def modify_connection(self):
-        return self.connection_update('modify')
+        status = self.connection_update('modify')
+        if self.edit_commands:
+            status = self.edit_connection()
+        return status
+
+    def edit_connection(self):
+        self.edit_commands += ['save', 'quit']
+        data = "\n".join(self.edit_commands)
+        self.edit_commands = []
```

Comment:
Why is `self.edit_commands` reset here?

Reply:
I suppose it's not really needed, but I would have to do something like this instead:
```diff
-        self.edit_commands += ['save', 'quit']
-        data = "\n".join(self.edit_commands)
+        data = "\n".join(self.edit_commands + ['save', 'quit'])
```
Since otherwise the `save` & `quit` commands will cause the `edit` command to quit if `edit_connection()` gets called again.

Reply:
Yeah, that would be better. The state manipulation as it is right now is more of a side-effect.